common/src/main/java/com/wynntils/wynn/handleditems/annotators/game/CraftedGearAnnotator.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright © Wynntils 2022.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.wynn.handleditems.annotators.game;
+
+import com.wynntils.handlers.item.ItemAnnotation;
+import com.wynntils.handlers.item.ItemAnnotator;
+import com.wynntils.mc.utils.ComponentUtils;
+import com.wynntils.utils.CappedValue;
+import com.wynntils.wynn.handleditems.items.game.CraftedGearItem;
+import com.wynntils.wynn.item.parsers.WynnItemMatchers;
+import com.wynntils.wynn.objects.Powder;
+import com.wynntils.wynn.objects.profiles.item.GearIdentification;
+import com.wynntils.wynn.utils.WynnUtils;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import net.minecraft.network.chat.Component;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.item.TooltipFlag;
+
+public final class CraftedGearAnnotator implements ItemAnnotator {
+    private static final Pattern CRAFTED_GEAR_PATTERN = Pattern.compile("^§3(.*)§b \\[100%\\]À$");
+
+    @Override
+    public ItemAnnotation getAnnotation(ItemStack itemStack) {
+        String name = ComponentUtils.getCoded(itemStack.getHoverName());
+        Matcher matcher = CRAFTED_GEAR_PATTERN.matcher(name);
+        if (!matcher.matches()) return null;
+
+        CappedValue durability = WynnItemMatchers.getDurability(itemStack);
+
+        List<GearIdentification> identifications = new ArrayList<>();
+        List<Powder> powders = List.of();
+
+        // Parse lore for identifications and powders
+        List<Component> lore = ComponentUtils.stripDuplicateBlank(itemStack.getTooltipLines(null, TooltipFlag.NORMAL));
+        lore.remove(0); // remove item name
```

Comment:
Can this crash with some weird 0 length lores?

Reply:
No, since it always contains the name as first line of the tooltip. (That's why we remove it). But I can make some defensive programming if we have interaction with bad mods.